File: Python/基础/字符串.py
<<<<<<< HEAD
#字符串用一对双引号或一对单引号来定义一个字符串
a = 10
print(a,type(a))    #10 <class 'int'>
b = '10'
print(b,type(b))    #10 <class 'str'>
#输出的结果相同，但是数值的类型是不同的

#字符串的操作
# + 拼接
a = 'Hello'
b = '408'
print(a + ' ' + b)

# *复制
a = '111'
print(a * 3)

# in/ not in 判断是否在字符串中返回bool值
a = 'Hello World'
print('e' in a )    #True
print('h' in b )    #False

#格式化输出
#1.占位符 %
age = 20
name = 'linx'
print('我的姓名是：%s,年龄是：%d'%(name,age))

str = 'abcde'
print('%10s'%(str))     #    abcde占十位向右对齐
print('%-10s!'%(str))   #abcde     !占十位向左对齐
print('%.2s'%(str))     #ab截取前面两位

number = 12.455
print('%.2f'%(number))  #%.2f表示保留两位小数，默认是保留六位

#format-----先占位，后填坑
print('{},{}'.format('linx',12))
number = 12.4111
print('{:.2f}'.format(number))  #保留两位小数

#f表达式
print(f'我的姓名是{name},我的年龄是{age}')
=======
#字符串用一对双引号或一对单引号来定义一个字符串
a = 10
print(a,type(a))    #10 <class 'int'>
b = '10'
print(b,type(b))    #10 <class 'str'>
#输出的结果相同，但是数值的类型是不同的

#字符串的操作
# + 拼接
a = 'Hello'
b = '408'
print(a + ' ' + b)

# *复制
a = '111'
print(a * 3)

# in/ not in 判断是否在字符串中返回bool值
a = 'Hello World'
print('e' in a )    #True
print('h' in b )    #False

#格式化输出
#1.占位符 %
age = 20
name = 'linx'
print('我的姓名是：%s,年龄是：%d'%(name,age))

str = 'abcde'
print('%10s'%(str))     #    abcde占十位向右对齐
print('%-10s!'%(str))   #abcde     !占十位向左对齐
print('%.2s'%(str))     #ab截取前面两位

number = 12.455
print('%.2f'%(number))  #%.2f表示保留两位小数，默认是保留六位

#format-----先占位，后填坑
print('{},{}'.format('linx',12))
number = 12.4111
print('{:.2f}'.format(number))  #保留两位小数

#f表达式
print(f'我的姓名是{name},我的年龄是{age}')
>>>>>>> 5169a62fdb26c2a597b35a0bc9f27034826ef6de

File: JavaScript/JS/Base/字符串.js
// 正常的字符串是用单引号或者双引号包裹使用
console.log('a')
console.log("a")
// 需要特殊处理时使用转义字符 \
console.log('a\nb')
console.log('a\ta')
console.log('\u4e2d') //unicode写法 \u####
console.log('\x41') //Ascll码

//多行字符串
var a = `
hello
world
111
`
console.log(a)

//模版字符串
var name = '张三'
var age = 20
var msg02 = `姓名：${name}, 年龄：${age}`

//字符串长度
var str = 'hello'
console.log(str.length)
//通过下标访问字符串
console.log(str[0])
//字符串的不可变性
str[0] = 'a'
console.log(str)
//大小写转化
str = str.toUpperCase() //大写
console.log(str)
str = str.toLowerCase() //小写
console.log(str)
//字符串截取，substring()，截取范围[)
var linx = 'asdfghjkl'
var linx02 = linx.substring(0, 3) //截取0到3的字符，不包含3
console.log(linx02)